api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -280,7 +333,19 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
   public ResponseEntity<Workspace> getWorkspace(String workspaceNamespace, String workspaceId) {
     org.pmiops.workbench.db.model.Workspace dbWorkspace = workspaceService.getRequired(
         workspaceNamespace, workspaceId);
-    return ResponseEntity.ok(TO_CLIENT_WORKSPACE.apply(dbWorkspace));
+    org.pmiops.workbench.firecloud.model.Workspace fcWorkspace;
+    try {
+      fcWorkspace = fireCloudService.getWorkspace(
+          workspaceNamespace, workspaceId).getWorkspace();
+    } catch (org.pmiops.workbench.firecloud.ApiException e) {
+      if (e.getCode() == 404) {
```

Comment:
Is a 404 returned if you don't have access to a workspace?

If instead it's a 403, we should probably handle that here?

Reply:
It is a 404 from FC.